File: fizzbuzz.java
class fizzbuzz{
    public static void main(String[] args){
        int n = Integer.parseInt(args[0]);
        for(int i=1; i<=n; i++){
            String o = i%3==0?"Fizz":"";
            o+=i%5==0?"Buzz":"";
            o+=o.length()==0?i:"";
            System.out.println(o);
        }
    }
}

File: fizzbuzz.exs
defmodule FizzBuzz do
  def fizzbuzz(n) do
    func = fn i -> 
      case [rem(i, 3), rem(i, 5), i] do
        [0, 0, _] -> "FizzBuzz"
        [0, _, _] -> "Fizz"
        [_, 0, _] -> "Buzz"
        [_, _, i] -> i
      end
    end
    import Enum
    1..n |> map(func) |> each(&IO.puts/1) 
  end
end

defmodule FizzBuzz2 do 
  def to(max) do
    1..max |> Enum.each fn(n) ->
      case {rem(n, 3) === 0, rem(n, 5) === 0} do
        {true, true} -> IO.puts "#{n} FizzBuzz"
        {true, false} -> IO.puts "#{n} Fizz"
        {false, true} -> IO.puts "#{n} Bazz"
        _ -> IO.puts n
      end
    end
  end
end


FizzBuzz.fizzbuzz(30)